TOOLS_OVERVIEW.md

```diff
@@ -0,0 +1,228 @@
+call environments, or memory snapshots. 
+
+Documentation on `goto-harness` can be found
+[here](http://cprover.diffblue.com/md__home_travis_build_diffblue_cbmc_doc_architectural_goto-harness.html)
+including details and examples.
+
+Details of all the options can be seen by running
+```
+goto-harness --help
+```
+
+
+## goto-instrument
+
+This is a collection of tools for analysing and modifying goto programs 
+(programs created with #goto-cc). Generally these take a goto program 
+and output another goto program.
+
+Further examples and documentation can be found:
+- [goto-instrument](http://cprover.diffblue.com/group__goto-instrument.html)
+has an overview of `goto-instrument` and a small tutorial example.
+- [Developer Tutorial](http://cprover.diffblue.com/tutorial.html) has high
+level overview and some example commands for `goto-instrument`.
+
+
+## janalyzer
+
+This provides a way to access and invoke various forms of analysis on
+Java programs.
+
+Documentation useful for this tool can be found:
+- [Analysis Information](http://cprover.diffblue.com/group__analyses.html)
+
+Details of all the options can be seen by running
+```
+janalyzer --help
+```
+
+## java-unit
+
+Runs Java unit tests. For more details use
+```
+java-unit --help
+```
+Default behaviour is to only show failed test cases.
+
+
+## jbmc
+
+This is the main analysis engine the performs the analysis
+of Java files using bounded model checking. This is a
+Java version of the `cbmc` tool (more documentation for
+`cbmc` is available and much of it applies to `jbmc`
+since they use many of the same back end utilities).
+
+For some light information on usage of the jbmc tool see the following documentation
+- [JBMC homepage](https://www.cprover.org/jbmc/) includes some
+simple examples and information.
+
+For details on how analysis is performed in the `cbmc` and
+`jbmc` tools see see the following documents:
+- [CBMC Architecture](http://cprover.diffblue.com/cbmc-architecture.html)
+gives a high level overview of the `cbmc` architecture and data flow that *should also apply to* `jbmc`.
+- [Background Concepts](http://cprover.diffblue.com/background-concepts.html)
+overviews all the key concepts used in the `jbmc` analysis.
+
+For details on compiling, testing, contributing, and documentation related to development see:
+- [CProver Development Documentation](http://cprover.diffblue.com/index.html)
+
+
+## jdiff
+
+Provides a variety of difference checks between two goto programs (produced
+by `goto-cc`). This invokes some of the `cbmc` tools to convert the goto
+program and then determine which functions are added/removed/changed.
+
+Details of all the options can be seen by running
+```
+jdiff --help
+```
+this includes both options for the difference, and options for the goto
+program instrumentation.
+
+
+## memory-analyzer
+
+This is a wrapper program that provides a front end to `gdb` that adds some
+useful features related to the other goto utilities. Note that to use
```

Comment:
It's a bit more than that.  `memory-analyzer` allows you to run an compiled binary, using `gdb` and when you are at the right point, it will build a harness that allows you to analyze the program /from that state/.

Reply:
Added text on this, hope it is clear(er) now.